### ISSUE TITLE
IR bug syntaxTests/*/base_contract.sol

### ISSUE THREAD

**Issue by hrkrshnn** [ethereum/solidity#10946]
```
contract B {
    function f() external {}
    function g() public {}
}
contract C is B {
    function h() public returns (bytes4 fs, bytes4 gs) {
        fs = B.f.selector;
        gs = B.g.selector;
        B.g();
    }
}
```

```
: Internal compiler error during compilation:
: /solidity/libsolidity/codegen/ir/IRGeneratorForStatements.cpp(1699): Throw in function virtual void solidity::frontend::IRGeneratorForStatements::endVisit(const solidity::frontend::MemberAccess&)
: Dynamic exception type: boost::wrapexcept<solidity::langutil::InternalCompilerError>
: std::exception::what: Invalid use of .selector: function ()
: [solidity::util::tag_comment*] = Invalid use of .selector: function ()
  
```

`solc --ir`

**Comment by axic** [comment]
Possibly duplicate of #10891

**Comment by bshastry** [comment]
Related to https://github.com/ethereum/solidity/issues/10891#issuecomment-775103029

**Comment by axic** [comment]
Fixed by #11015.
